code/createPowerTable.m

```diff
@@ -0,0 +1,57 @@
+function [powerTable]  = createPowerTable(folderPath)
+%%
+% Function to unravel Power data
+% Input:
+%   folderPath: path to the Device* folder, which contains all the .json
+%   files for this recording
+%
+% Output: powerTable (power data in table format, without band limits
+% decoded)
+%%
+% Load power data
+rawPowerData = jsondecode(fixMalformedJson(fileread([folderPath filesep 'RawDataPower.json']),'EventLog'));
+
+% Initalize output
+powerTable = table();
```

Comment:
Since you already know the variables you want to extract and how large of a data set it will be, I would preallocate the struct to make the code run faster. I'm not 100% sure a struct works like an array, but assigning a new field may make MATLAB find a new memory space to hold all the old data and the new data, and then gets rid of the variable data at the old location. I noted this simple function runs slower than I would expect.

Reply:
in general, the efficiency (for speed) of the code is a non-issue since we only have ~5K sessions max after 2 years and 10 patients and reading all the data with this code (or similar) to construct database takes under 10 minutes. So adding one file (even with this very inefficient code) is not a big deal. The bigger problems arise from reading the time domain data from the very large .json's. This is mostly solves with Duke's Turtle Json but should eventually (Turtle Json) be adjusted to allow run on Linux and more modern version of Matlab.